zookeeper-server/src/test/java/org/apache/zookeeper/server/backup/storage/impl/FileSystemBackupStorageTest.java

```diff
@@ -0,0 +1,144 @@
+  private static List<File> testFiles;
+  private static List<File> backupFiles;
+  private static final String NAMESPACE = "test";
+
+  @BeforeClass
+  public static void beforeClass() throws ConfigException, IOException {
+    // Build backup config that point the backup storage to a local directory
+    BackupConfig backupConfig = new BackupConfig.Builder().setEnabled(true)
+        .setStatusDir(new File("./localBackupStorageTest/backup/status")).
+            setTmpDir(new File("./localBackupStorageTest/tmp/backup"))
+        .setStorageProviderClassName(FileSystemBackupStorage.class.getName())
+        .setBackupStoragePath("./localBackupStorageTest").setNamespace(NAMESPACE).build().get();
+
+    backupStorage = new FileSystemBackupStorage(backupConfig);
+    testFolderPath =
+        Paths.get(Paths.get("").toAbsolutePath().toString(), backupConfig.getBackupStoragePath())
+            .toString();
+    backupFileDirPath = String.join(File.separator, testFolderPath, NAMESPACE);
+    testFileDirPath = String.join(File.separator, testFolderPath, "testFiles");
+    testFiles = new ArrayList<>();
+    backupFiles = new ArrayList<>();
+
+    for (String testFileName : testFileNames) {
+      String testFilePath = String.join(File.separator, testFileDirPath, testFileName);
+      File testFile = new File(testFilePath);
+      testFiles.add(testFile);
+      BackupStorageUtil.createFile(testFile, false);
+      FileWriter fileWriter = new FileWriter(testFile);
+      for (int i = 0; i < 1000; i++) {
+        fileWriter.write(testFileName);
+        fileWriter.write('\n');
+      }
+      fileWriter.close();
+      Assert.assertTrue(testFile.exists());
+      Assert.assertTrue(sizeOf(testFile) > 0);
+    }
+  }
+
+  @AfterClass
+  public static void afterClass() throws IOException {
+    FileUtils.deleteDirectory(new File(testFolderPath));
+  }
+
+  @Test
+  public void test0_CopyToBackupStorage() throws IOException {
+    for (File testFile : testFiles) {
+      backupStorage.copyToBackupStorage(testFile, testFile);
+      String backupFilePath =
+          BackupStorageUtil.constructBackupFilePath(testFile.getName(), backupFileDirPath);
+      File backupFile = new File(backupFilePath);
+      backupFiles.add(backupFile);
+      Assert.assertEquals(sizeOf(testFile), sizeOf(backupFile));
+    }
+  }
+
+  @Test
+  public void test1_GetBackupFileInfo() throws IOException {
+    BackupFileInfo validFileInfo = backupStorage.getBackupFileInfo(testFiles.get(0));
+    BackupFileInfo invalidFileInfo =
+        backupStorage.getBackupFileInfo(new File("log.invalidBackupFile"));
+    Assert.assertEquals(validFileInfo.getSize(), sizeOf(testFiles.get(0)));
+    Assert.assertEquals(invalidFileInfo.getSize(), BackupFileInfo.NOT_SET);
+  }
+
+  @Test
+  public void test2_GetBackupFileInfos() throws IOException {
+    List<BackupFileInfo> fileInfos = backupStorage.getBackupFileInfos(new File("/log"), "log");
+    Assert.assertEquals(2, fileInfos.size());
```

Comment:
Nit: should the order be reversed? (actual, expected)?

Reply:
Seems this is another difference between junit and TestNG. In junit it's like this: public static void assertEquals(long expected, long actual).